sdk/core/azure-core-amqp/src/models/message_target.cpp

```diff
@@ -670,4 +670,11 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models { namespace
     os << "}";
     return os;
   }
+
+  Azure::Core::Amqp::_detail::AmqpTargetImplementation* AmqpTargetFactory::ToImplementation(
+      Models::_internal::MessageTarget const& target)
+  {
+    return target.m_impl->m_target.get();
```

Comment:
Would this ever be null? If not, should it return a reference instead of a pointer?

Reply:
`AmqpTargetImplementation` is a declared but undefined type, and this function returns a pointer to that declared but undefined type.

In practice, it cannot be null.

But more importantly, it *cannot* be returned as a reference - it's basically a strongly typed `void*`.

This follows the pattern of defining foreign types as undefined types.

Reply:
Not arguing against the pointer or for the reference, but FWIW you can use references to forward-declared types, unless, of course, the type is a typedef of `void` - `void&` are not possible.

Reply:
In this case I specifically *don't* want a reference. Whenever I'm returning a raw pointer, the pointer is owned by some other type and the presence of the raw pointer is a strong message that someone else owns the type.

Returning a reference to an opaque type seems like it's the wrong semantics, especially when the opaque type is contained within a `std::unique_ptr` (in which case it's *hard* to get a reference to the underlying pointer and trivial to get a pointer to the underlying pointer).